apps/ejabberd/src/ejabberd_admin.erl

```diff
@@ -325,6 +337,86 @@ registered_users(Host) ->
     SUsers = lists:sort(Users),
     lists:map(fun({U, _S}) -> U end, SUsers).
 
+-spec import_users(Filename :: string()) -> [{ok, ejabberd:user()} |
+                                             {exists, ejabberd:user()} |
+                                             {not_allowed, ejabberd:user()} |
+                                             {invalid_jid, ejabberd:user()} |
+                                             {null_password, ejabberd:user()} |
+                                             {bad_csv, binary()}].
+import_users(File) ->
+    FileStream = stdio:file(File),
+    CsvStream = csv:stream(FileStream),
+    Workers = spawn_link_workers(),
+    WorkersQueue = queue:from_list(Workers),
```

Comment:
This looks like a worker pool written from scratch. How about using https://github.com/inaka/worker_pool for that? We already have it in some places in MongooseIM.